### PR TITLE
Package rename

### PR DESCRIPTION
**Pull request by Reettaphant**
Allow for editing the names of story packages. 

screen shot 2016-03-09 at 09 51 40

### CODE REVIEW

File: app/services/Database.scala [hunk 1]
```diff
@@ -96,17 +96,33 @@ object Database {
...
     WithExceptionHandling(errorMessage, {
       val modifyDate = new DateTime().withZone(DateTimeZone.UTC)
 
-      val updateSpec = new UpdateItemSpec()
-        .withPrimaryKey("id", id)
-        .addAttributeUpdate(new AttributeUpdate("lastModify").put(modifyDate.toString))
-        .addAttributeUpdate(new AttributeUpdate("lastModifyBy").put(user.email))
-        .addAttributeUpdate(new AttributeUpdate("lastModifyByName").put(user.fullName))
-        .withReturnValues(ReturnValue.ALL_NEW)
+      val updateSpec = newName match {
```
Comment (piuccio) [inline app/services/Database.scala]: you can avoid duplication leaving the original code and then

```
if (newName.nonEmpty) {
   updateSpec.addAttributeUpdate(new AttributeUpdate("packageName").put(name))
}
```

File: public/src/js/widgets/columns/packages.js [hunk 2]
```diff
@@ -121,16 +122,56 @@ export default class Package extends ColumnWidget {
...
+            type: 'post',
+            data: JSON.stringify({
+                name: name,
+                isHidden: parent.baseModel.priority === 'training'
+            })
+        })
+        .then((response) => {
+
+            var storyPackage = new StoryPackage(response);
+            var results = parent.searchResults();
```
Comment (piuccio) [inline public/src/js/widgets/columns/packages.js]: I don't like the logic here, children shouldn't change things on `parent.`. It's better if the parent listens to `update:package` and updates the search results in the callback

File: public/src/js/widgets/columns/packages.js [hunk 3]
```diff
@@ -121,16 +122,56 @@ export default class Package extends ColumnWidget {
...
+            results[index] = storyPackage;
+            storyPackage.meta.lastModifyHuman(humanTime(storyPackage.meta.lastModify()));
+            storyPackage.savedDisplayName = storyPackage.meta.name();
+            parent.searchResults(results);
+            mediator.emit('update:package', response);
+            storyPackage.editing(false);
+        });
```
Comment (piuccio) [inline public/src/js/widgets/columns/packages.js]: you should handle exceptions somehow

File: public/src/js/widgets/columns/packages.js (hunk 2 above)
Comment (Reettaphant) [inline public/src/js/widgets/columns/packages.js]: This code was really confusing, I've updated the function context.

File: app/services/Database.scala [hunk 4]
```diff
@@ -96,17 +96,21 @@ object Database {
...
     WithExceptionHandling(errorMessage, {
       val modifyDate = new DateTime().withZone(DateTimeZone.UTC)
 
       val updateSpec = new UpdateItemSpec()
-        .withPrimaryKey("id", id)
-        .addAttributeUpdate(new AttributeUpdate("lastModify").put(modifyDate.toString))
-        .addAttributeUpdate(new AttributeUpdate("lastModifyBy").put(user.email))
-        .addAttributeUpdate(new AttributeUpdate("lastModifyByName").put(user.fullName))
-        .withReturnValues(ReturnValue.ALL_NEW)
+            .withPrimaryKey("id", id)
```
Comment (piuccio) [inline app/services/Database.scala]: can you check the indentation?

File: public/src/js/widgets/columns/packages.js [hunk 5]
```diff
@@ -117,22 +118,71 @@ export default class Package extends ColumnWidget {
...
+                name: name,
+                isHidden: this.baseModel.priority === 'training'
+            })
+        })
+        .then((response) => {
+
+            var storyPackage = new StoryPackage(response);
+            var results = this.searchResults();
+            results[index] = storyPackage;
+            storyPackage.meta.lastModifyHuman(humanTime(storyPackage.meta.lastModify()));
```
Comment (piuccio) [inline public/src/js/widgets/columns/packages.js]: this and the following line should be done in the story package controller

File: public/src/js/widgets/columns/packages.js [hunk 6]
```diff
@@ -152,14 +202,13 @@ function performSearch(searchTerm) {
 function displayResults(results = {}) {
     if (this.searchInProgress()) {
         this.searchResults((results || []).map(result => {
-            return Object.assign({
-                lastModifyHuman: humanTime(new Date(result.lastModify))
-            }, result);
+            var storyPackage = new StoryPackage(result);
+            storyPackage.meta.lastModifyHuman(humanTime(new Date(result.lastModify)));
```
Comment (piuccio) [inline public/src/js/widgets/columns/packages.js]: this line should be done in the story package constructor

File: public/src/js/widgets/columns/story-package.js [hunk 7]
```diff
@@ -52,8 +52,29 @@ export default class Front extends ColumnWidget {
             }
         });
 
-        this.listenOn(mediator, 'find:package', function(storyPackage) {
+        this.listenOn(mediator, 'update:package', function(storyPackage) {
             var existingPackages = this.baseModel.latestPackages();
+            var index = _.findIndex(existingPackages, existingPackage => existingPackage.id === storyPackage.id);
+            if (index !== -1) {
+              existingPackages[index] = storyPackage;
```
Comment (piuccio) [inline public/src/js/widgets/columns/story-package.js]: indentation

**Comment by piuccio** [comment]
can you also change your `var` into `const` please?

File: public/src/js/widgets/columns/packages.js [hunk 8]
```diff
@@ -153,15 +194,12 @@ function performSearch(searchTerm) {
 function displayResults(results = {}) {
     if (this.searchInProgress()) {
         this.searchResults((results || []).map(result => {
```
Comment (piuccio) [inline public/src/js/widgets/columns/packages.js]: you could write this in one line `.map(result => new StoryPackage(result))`

**Comment by piuccio** [comment]
:+1:

**Pull request merged**